sympy/solvers/tests/test_solvers.py

```diff
@@ -1625,6 +1578,60 @@ def test_other_lambert():
     assert set(solve(3**cos(x) - cos(x)**3)) == set(
         [acos(3), acos(-3*LambertW(-log(3)/3)/log(3))])
 
+@slow
+def test_lambert_bivariate():
+    # tests passing current implementation
+    from sympy.solvers.bivariate import _filtered_gens, _solve_lambert
+    assert solve((a/x + exp(x/2)).diff(x), x) == \
+            [4*LambertW(-sqrt(2)*sqrt(a)/4), 4*LambertW(sqrt(2)*sqrt(a)/4)]
+    assert solve(x*log(x) + 3*x + 1, x) == \
+            [exp(-3 + LambertW(-exp(3)))]
```

Comment:
In this the l in bivariate comes out to be `LambertW(_u*exp(3), -1)` so because of positive assumption of `_u` l.is_real is returning False.

Reply:
Clearly the assumptions in sympy are getting `_u*exp(3)` to be positive and where as it know that `W-1` gives real solution when args is between `-1/e to 0`. So `l.is_real` returns `false`. I don't think we can do here something because if we avoid assumptions while calculating this then more wrong solutions will be included. Please help here.

Reply:
And moreover `W-1` solution here is complex one which i think we are not covering here.
```python
>>>N(exp(-3 + LambertW(-exp(3), -1)))
-0.216986323494879 - 0.257692008289364*I
```

Reply:
Yeah, it's fine then. It should not be considered as of now if it is not real.

Reply:
`solve` doesn't have a bias against solutions with imaginary parts; it's ok to return such solutions.